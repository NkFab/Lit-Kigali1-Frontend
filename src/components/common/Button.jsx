import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const {
    label, className, disable, onClick,
  } = props;
  return (
    <button type="submit" onClick={onClick} className={className} disabled={disable}>
      {label}
    </button>
  );
};

const propTypes = {
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  disable: PropTypes.bool,
  className: PropTypes.string.isRequired,
};
const defaultProps = {
  disable: false,
};
Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
export { Button };
