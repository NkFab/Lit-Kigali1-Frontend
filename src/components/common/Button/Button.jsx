import React from 'react';
import PropTypes from 'prop-types';

const Button = ({
  children, type, classes, onClick, disabled,
}) => (
  <button disabled={disabled} type={type} className={`button ${classes}`} onClick={onClick}>
    {children}
  </button>
);

Button.propTypes = {
  disabled: PropTypes.string,
  type: PropTypes.string,
  children: PropTypes.any.isRequired,
  classes: PropTypes.string,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  disabled: '',
  type: 'button',
  classes: '',
  onClick: () => 'button',
};

export default Button;
