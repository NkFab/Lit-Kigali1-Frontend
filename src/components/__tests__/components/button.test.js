import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../common/Button';

describe('<Button/>', () => {
  let btn;
  const Callback = jest.fn();

  beforeEach(() => {
    btn = shallow(
      <Button type="submit" onClick={Callback} className={{ color: 'blue' }} disabled={false}>
        Ok
      </Button>,
    );
  });

  test('`Button` component should be rendered', () => {
    expect(btn).toMatchSnapshot();
  });

  test('Button should be clicked', () => {
    btn.find('btn').simulate('click');
    expect(Callback).toHaveBeenCalled();
  });
});
