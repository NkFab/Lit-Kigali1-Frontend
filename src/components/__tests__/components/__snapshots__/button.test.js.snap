// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> \`Button\` component should be rendered 1`] = `ShallowWrapper {}`;
